src/_P013_HCSR04.ino

```diff
@@ -83,10 +82,11 @@ boolean Plugin_013(byte function, struct EventStruct *event, String& string)
     case PLUGIN_INIT:
       {
         Plugin_013_init = true;
-        pinMode(Settings.TaskDevicePin1[event->TaskIndex], OUTPUT);
-        pinMode(Settings.TaskDevicePin2[event->TaskIndex], INPUT_PULLUP);
+
+        Plugin_013_TRIG_Pin = Settings.TaskDevicePin1[event->TaskIndex];
         Plugin_013_IRQ_Pin = Settings.TaskDevicePin2[event->TaskIndex];
-        attachInterrupt(Settings.TaskDevicePin2[event->TaskIndex], Plugin_013_interrupt, CHANGE);
+
+        sonar = new NewPing(Plugin_013_TRIG_Pin, Plugin_013_IRQ_Pin);
```

Comment:
Still needs something like this before assigning the `new NewPing`, since init can be called more than once.
```C++
if (sonar)
{
  delete sonar;
  sonar=NULL;
}
```